tsl/src/dist_util.c

```diff
@@ -338,3 +310,66 @@ dist_util_is_compatible_version(const char *data_node_version, const char *acces
 
 	return (data_node_major == access_node_major) && (data_node_minor <= access_node_minor);
 }
+
+/*
+ * Returns results from SRFs on remote nodes
+ * Pass the fcinfo information from the original PG function
+ * args: node_name is NAME and sql_query is of TEXT type
+ */
+static Datum
+dist_util_remote_srf_query(FunctionCallInfo fcinfo, const char *node_name, const char *sql_query)
+{
+	FuncCallContext *funcctx;
+	PGresult *result;
+
+	Assert(node_name != NULL && sql_query != NULL);
+	if (SRF_IS_FIRSTCALL())
+	{
+		MemoryContext oldcontext;
+		TupleDesc tupdesc;
+
+		funcctx = SRF_FIRSTCALL_INIT();
+		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+
+		if (get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("function returning record called in context "
+							"that cannot accept type record")));
+
+		funcctx->user_fctx =
+			ts_dist_cmd_invoke_on_data_nodes(sql_query, list_make1((void *) node_name), true);
```

Comment:
Cast to `void*` not needed (and can hide errors).
```suggestion
			ts_dist_cmd_invoke_on_data_nodes(sql_query, list_make1(node_name), true);
```

Reply:
needed , otherwise compiler errors.